x/posts/internal/types/codec.go

```diff
@@ -13,8 +13,13 @@ func init() {
 
 // RegisterCodec registers concrete types on the Amino codec
 func RegisterCodec(cdc *codec.Codec) {
-	cdc.RegisterConcrete(MsgCreatePost{}, "desmos/MsgCreatePost", nil)
+	cdc.RegisterInterface((*MsgCreatePost)(nil), nil)
+	cdc.RegisterConcrete(MsgCreateTextPost{}, "desmos/MsgCreateTextPost", nil)
+	cdc.RegisterConcrete(MsgCreateMediaPost{}, "desmos/MsgCreateMediaPost", nil)
 	cdc.RegisterConcrete(MsgEditPost{}, "desmos/MsgEditPost", nil)
 	cdc.RegisterConcrete(MsgAddPostReaction{}, "desmos/MsgAddPostReaction", nil)
 	cdc.RegisterConcrete(MsgRemovePostReaction{}, "desmos/MsgRemovePostReaction", nil)
+	cdc.RegisterInterface((*Post)(nil), nil)
```

Comment:
What happens if you don't register the interface or concrete types? Can we test this?

Reply:
How can I test such a thing? building custom codecs without it?

Reply:
@bragaz Just delete those lines and run the tests. If they pass, it means they are useless.

Reply:
@bragaz Have you tried deleting them? What is the result?